native-asset/native-asset-contract/src/main.sw

```diff
@@ -258,23 +292,58 @@ impl SRC3 for Contract {
     /// # Examples
     ///
     /// ```sway
-    /// use src3::SRC3;
+    /// use standards::src3::SRC3;
+    /// use std::constants::DEFAULT_SUB_ID;
     ///
-    /// fn foo(contract: ContractId, asset_id: AssetId) {
+    /// fn foo(contract: ContractId) {
     ///     let contract_abi = abi(SR3, contract);
+    ///     let asset_id = AssetId::new(contract, DEFAULT_SUB_ID);
     ///     contract_abi {
     ///         gas: 10000,
     ///         coins: 100,
     ///         asset_id: AssetId,
-    ///     }.burn(ZERO_B256, 100);
+    ///     }.burn(DEFAULT_SUB_ID, 100);
     /// }
     /// ```
+    #[payable]
     #[storage(read, write)]
     fn burn(sub_id: SubId, amount: u64) {
+        require(msg_amount() == amount, AmountError::AmountMismatch);
```

Comment:
Can we just use `msg_amount()` directly and avoid the extra require?

Reply:
As per the spec of SRC3 the argument is required